Dockerfile

```diff
@@ -1,87 +1,49 @@
 
+ENV LOG_LEVEL=trace
+ENV DEBIAN_FRONTEND=noninteractive
+
 # -----------------------------------------------
 # --- Install Basic Software --------------------
 # -----------------------------------------------
 
-RUN \
-  apt-get -qq update && \
-  apt-get -qq install apt-utils 2>/dev/null && \
-  apt-get -qq dist-upgrade && \
-  echo "applying workaround for ubuntu/postfix bug described in https://github.com/docker-mailserver/docker-mailserver/issues/2023#issuecomment-855326403" && \
-  mv /bin/hostname{,.bak} && \
-  echo "echo docker-mailserver.invalid" > /bin/hostname && \
-  chmod +x /bin/hostname && \
-  apt-get -qq install postfix && \
-  mv /bin/hostname{.bak,} && \
-  apt-get -qq --no-install-recommends install \
-  # A - D
-  altermime amavisd-new apt-transport-https arj binutils bzip2 bsd-mailx \
-  ca-certificates cabextract clamav clamav-daemon cpio curl \
-  dbconfig-no-thanks dovecot-core dovecot-fts-xapian dovecot-imapd \
-  dovecot-ldap dovecot-lmtpd dovecot-managesieved dovecot-pop3d \
-  dovecot-sieve dovecot-solr dumb-init \
-  # E - O
-  ed fetchmail file gamin gnupg gzip iproute2 \
-  locales logwatch lhasa libdate-manip-perl libldap-common liblz4-tool \
-  libmail-spf-perl libnet-dns-perl libsasl2-modules lrzip lzop \
-  netcat-openbsd nftables nomarch opendkim opendkim-tools opendmarc \
-  # P - Z
-  pax pflogsumm postgrey p7zip-full postfix-ldap postfix-pcre \
-  postfix-policyd-spf-python postsrsd pyzor \
-  razor rpm2cpio rsyslog sasl2-bin spamassassin supervisor \
-  unrar-free unzip uuid whois xz-utils && \
-  # Fail2Ban
-  gpg --keyserver ${FAIL2BAN_GPG_PUBLIC_KEY_SERVER} \
-    --recv-keys ${FAIL2BAN_GPG_PUBLIC_KEY_ID} 2>&1 && \
-  curl -Lkso fail2ban.deb ${FAIL2BAN_DEB_URL} && \
-  curl -Lkso fail2ban.deb.asc ${FAIL2BAN_DEB_ASC_URL} && \
-  FINGERPRINT=$(LANG=C gpg --verify \
-  fail2ban.deb.asc fail2ban.deb 2>&1 \
-    | sed -n 's#Primary key fingerprint: \(.*\)#\1#p') && \
-  if [[ -z ${FINGERPRINT} ]]; then \
-    echo "ERROR: Invalid GPG signature!" >&2; exit 1; fi && \
-  if [[ ${FINGERPRINT} != "${FAIL2BAN_GPG_FINGERPRINT}" ]]; then \
-    echo "ERROR: Wrong GPG fingerprint!" >&2; exit 1; fi && \
-  dpkg -i fail2ban.deb 2>&1 && \
-  rm fail2ban.deb fail2ban.deb.asc && \
-  # cleanup
-  apt-get -qq autoremove && \
-  apt-get -qq autoclean && \
-  apt-get -qq clean && \
-  rm -rf /var/lib/apt/lists/* && \
-  c_rehash 2>&1
-
-COPY ./target/scripts/helpers/log.sh /usr/local/bin/helpers/log.sh
-COPY ./target/bin/sedfile /usr/local/bin/sedfile
-
-RUN chmod +x /usr/local/bin/sedfile
+COPY target/scripts/build/* /build/
+COPY target/scripts/helpers/log.sh /usr/local/bin/helpers/log.sh
+RUN /build/packages.sh
+
+COPY target/bin/sedfile /usr/local/bin/sedfile
 
 # -----------------------------------------------
 # --- ClamAV & FeshClam -------------------------
 # -----------------------------------------------
 
+# hadolint ignore=DL3021
+COPY --link --from=stage-clamav /var/lib/clamav /var/lib/clamav
```

Comment:
I think, copying the database from the official image is a good idea, instead of running `freshclam` on our own. Running freshclam is rate limited. I didn't check it, if that was already the case in the past, but theoretically we can end with an image without a current DB due to rate limits by clamav. At least while testing/building locally, I run very quickly in that rate limit.

Reply:
Nice observation. I will see what I can do.

Reply:
It should be as simple as this:

`COPY --link --from=clamav/clamav:latest /var/lib/clamav /var/lib/clamav`